### ISSUE TITLE
Setting theme/colors from code is not consistent in rc 3.0

### ISSUE THREAD

**Issue by Al-Dyachkov** [AvaloniaCommunity/Material.Avalonia#162]
Steps to reproduce :
Add color changes  to Material.Demo following way
```C#
        public static void UseMaterialUIDarkTheme() {
            MaterialThemeStyles.BaseTheme    = BaseThemeMode.Dark;
            MaterialThemeStyles.PrimaryColor = PrimaryColor.Green;
            MaterialThemeStyles.SecondaryColor = SecondaryColor.Green;
        }
        public static void UseMaterialUILightTheme() {
            MaterialThemeStyles.BaseTheme      = BaseThemeMode.Light;
            MaterialThemeStyles.PrimaryColor   = PrimaryColor.Red;
            MaterialThemeStyles.SecondaryColor = SecondaryColor.Red;
        }
```
Run & switch themes several times.
Color & theme are not applying consistently

**Comment by SKProCH** [comment]
Hello, @Al-Dyachkov 

I tried to reproduce it but couldn't:
![Animation](https://user-images.githubusercontent.com/29896317/159126525-0c84f1d8-9210-490d-bbae-d8c742cb5a7e.gif)
As a result, the state from the last click is always applied.  

Intermediate values ​​do not always change, yes. This is due to the fact that in fact the change theme operation is quite resource intensive and is done by many calls to the dispatcher to change each color. As a result, we got rid of the freeze when changing the theme, and masked the duration of the theme change with animation. But since changing the theme is actually still quite a long operation, if you quickly click on the theme change buttons, the theme will not change every time, but only after the last click.

So, i cant reproduce you error, or this is actually not a error but a limitation, as i described above.

**Comment by Al-Dyachkov** [comment]
This is how it works for me, built from latest master commit (a6459717adc74f26fb18bb5b46657aa92ae65953)
Recorded on Windows 10, tested on Linux as well.
https://user-images.githubusercontent.com/10960735/159128243-136081e4-2752-45e3-b496-fabfbdf3b56d.mp4

**Comment by Al-Dyachkov** [comment]
I've tested simple method to change all parameters at once.
It seems to fix this issue with changing multiple parameters at once.
I can add parameter validation & make a PR.

```code C#
 public void Modify( BaseThemeMode  Mode,
                            PrimaryColor   Primary,
                            SecondaryColor Secondary )
        {
            var theme = _theme; 
            
            theme             = theme.SetBaseTheme(Mode.GetBaseTheme());
            theme             = theme.SetPrimaryColor(SwatchHelper.Lookup[(MaterialColor)Primary]);
            theme             = theme.SetSecondaryColor(SwatchHelper.Lookup[(MaterialColor)Secondary]);
            
            _theme            = theme;
            this.CurrentTheme = theme;
        }
```

**Comment by SKProCH** [comment]
Your code does not include properties update. 
Ill try to write more accurate fix.

**Comment by Al-Dyachkov** [comment]
>Ill try to write more accurate fix.

Well, OK. This was a quick fix, after all.
While you are on it :

private bool _isBaseThemePropertyApplied;
private bool _isPrimaryColorPropertyApplied;
private bool _isSecondaryColorPropertyApplied;

these fields are only set to true and never reset. Maybe they were meant to be local variables ?

**Comment by SKProCH** [comment]
> these fields are only set to true and never reset. Maybe they were meant to be local variables ?
This is a workaround for a initial values. When Avalonia create instance of MaterialTheme `OnPropertyChanged` will be called 3 times (for setting initial value for `BaseThemeProperty`, `PrimaryColorProperty` and `SecondaryColorProperty`). This properties will prevent calling theme change 3 times.

**Comment by SKProCH** [comment]
@Al-Dyachkov can you try code from #165? Seems like this fixed problem for me.  
Also, this is an draft, if the fix works, then I will complete the changes in the future

**Comment by Al-Dyachkov** [comment]
Yep, throttling works.

**Comment by SKProCH** [comment]
Fixed in `3.0.0-avalonia10-preview2`
